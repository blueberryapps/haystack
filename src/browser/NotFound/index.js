import React, { PropTypes as RPT } from 'react';

const NotFound = ({ location: { pathname }, staticContext }) => {
  staticContext.status = 404; // eslint-disable-line no-param-reassign

  return (
    <div>
      <h1>404: Needle {pathname} not found in Haystack :(</h1>
      You can redirect to <a href="/">home</a>.
    </div>
  );
};

NotFound.propTypes = {
  location: RPT.shape({
    pathname: RPT.string.isRequired
  }).isRequired,
  staticContext: RPT.shape({
    status: RPT.number
  })
};

NotFound.defaultProps = {
  staticContext: {}
};


export default NotFound;
