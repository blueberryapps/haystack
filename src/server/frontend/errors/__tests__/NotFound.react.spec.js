import NotFound from '../NotFound.react';
import { shallow } from 'enzyme';
import React from 'react';

test('shallowly renders NotFound', () => {
  expect(shallow(<NotFound path="whatever" />)).toMatchSnapshot();
});
