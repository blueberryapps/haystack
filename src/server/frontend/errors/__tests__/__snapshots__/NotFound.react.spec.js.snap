exports[`test shallowly renders NotFound 1`] = `
<div>
  <h1>
    404: Needle (
    whatever
    ) not found in Haystack :(
  </h1>
  You can redirect to 
  <a
    href="/">
    home
  </a>
  .
</div>
`;
